pkg/appgw/health_probes.go

```diff
@@ -105,6 +106,9 @@ func (c *appGwConfigBuilder) generateHealthProbe(backendID backendIdentifier) *n
 		}
 	}
 
+	if probe.Path != nil {
+		probe.Path = to.StringPtr(strings.TrimRight(*probe.Path, "*"))
```

Comment:
We should add a unit test to illustrate what's going on here...

Reply:
Will address this!